boost v1.71 - Program fails to compile after including header : <boost/process.hpp>

I dont't know if it is relatet to #92, but i can't compile the same source code with only the header included with boost 1.71.

```
In file included from W:/boost_1_71_0/boost/process/detail/windows/handles.hpp:11,
                 from W:/boost_1_71_0/boost/process/detail/used_handles.hpp:17,
                 from W:/boost_1_71_0/boost/process/detail/windows/async_in.hpp:20,
                 from W:/boost_1_71_0/boost/process/async.hpp:49,
                 from W:/boost_1_71_0/boost/process.hpp:23,
                 from X:\trunk\test\test_boost_process.cpp:1:
W:/boost_1_71_0/boost/process/detail/windows/handle_workaround.hpp:209:51: error: expected ')' before '*' token
 typedef ::boost::winapi::NTSTATUS_ (__kernel_entry *nt_system_query_information_p )(
                                    ~              ^~
                                                   )
W:/boost_1_71_0/boost/process/detail/windows/handle_workaround.hpp:224:51: error: expected ')' before '*' token
 typedef ::boost::winapi::NTSTATUS_ (__kernel_entry *nt_query_object_p )(
                                    ~              ^~
                                                   )
W:/boost_1_71_0/boost/process/detail/windows/handle_workaround.hpp: In function 'boost::winapi::NTSTATUS_ boost::process
::detail::windows::workaround::nt_system_query_information(boost::process::detail::windows::workaround::SYSTEM_INFORMATI
ON_CLASS_, void*, boost::winapi::ULONG_, boost::winapi::PULONG_)':
W:/boost_1_71_0/boost/process/detail/windows/handle_workaround.hpp:240:12: error: 'nt_system_query_information_p' does n
ot name a type; did you mean 'nt_system_query_information'?
     static nt_system_query_information_p f = reinterpret_cast<nt_system_query_information_p>(::boost::winapi::get_proc_
address(h, "NtQuerySystemInformation"));
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            nt_system_query_information
...

--- a/include/boost/process/detail/windows/handle_workaround.hpp
+++ b/include/boost/process/detail/windows/handle_workaround.hpp
@@ -198,20 +198,20 @@ typedef struct _OBJECT_TYPE_INFORMATION_ {
 
 
 /*
-__kernel_entry NTSTATUS NtQuerySystemInformation(
+NTSTATUS NtQuerySystemInformation(
   IN SYSTEM_INFORMATION_CLASS SystemInformationClass,
   OUT PVOID                   SystemInformation,
   IN ULONG                    SystemInformationLength,
   OUT PULONG                  ReturnLength
 );
  */
-typedef ::boost::winapi::NTSTATUS_ (__kernel_entry *nt_system_query_information_p )(
+typedef ::boost::winapi::NTSTATUS_ (*nt_system_query_information_p )(
         SYSTEM_INFORMATION_CLASS_,
         void *,
         ::boost::winapi::ULONG_,
         ::boost::winapi::PULONG_);
 /*
-__kernel_entry NTSYSCALLAPI NTSTATUS NtQueryObject(
+NTSYSCALLAPI NTSTATUS NtQueryObject(
   HANDLE                   Handle,
   OBJECT_INFORMATION_CLASS ObjectInformationClass,
   PVOID                    ObjectInformation,
@@ -220,7 +220,7 @@ __kernel_entry NTSYSCALLAPI NTSTATUS NtQueryObject(
 );
  */
 
-typedef ::boost::winapi::NTSTATUS_ (__kernel_entry *nt_query_object_p )(
+typedef ::boost::winapi::NTSTATUS_ (*nt_query_object_p )(
         ::boost::winapi::HANDLE_,
         OBJECT_INFORMATION_CLASS_,
         void *,

--- a/include/boost/process/detail/windows/handles.hpp
+++ b/include/boost/process/detail/windows/handles.hpp
@@ -136,7 +136,7 @@ struct limit_handles_ : handler_base_ext
                 [&](::boost::winapi::HANDLE_ handle)
                 {
                     auto itr = std::find(all_handles.begin(), all_handles .end(), handle);
-                    DWORD flags = 0u;
+                    ::boost::winapi::DWORD_ flags = 0u;
                     if (itr != all_handles.end())
                         *itr = ::boost::winapi::INVALID_HANDLE_VALUE_;
                     else if ((::boost::winapi::GetHandleInformation(*itr, &flags) != 0)

